basyx/aas/adapter/http.py

```diff
@@ -569,7 +589,16 @@ def _get_submodel_reference(cls, aas: model.AssetAdministrationShell, submodel_i
                 return ref
         raise NotFound(f"The AAS {aas!r} doesn't have a submodel reference to {submodel_id!r}!")
 
-    def _get_shells(self, request: Request) -> Iterator[model.AssetAdministrationShell]:
+    @classmethod
+    def _get_slice(cls, request: Request, iterator: Iterator) -> Tuple[Iterator, int]:
```

Comment:
We can even use a type variable here to indicate, that the returned iterator is of the same type as the one passed to the function, i.e. `Iterator[T]`. Function looks good to me otherwise!